aws-datastore/src/main/java/com/amplifyframework/datastore/syncengine/PersistentMutationOutbox.java

```diff
@@ -183,16 +242,24 @@ private Completable removeNotLocking(@NonNull TimeBasedUuid pendingMutationId) {
     public Completable load() {
         return Completable.create(emitter -> {
             inFlightMutations.clear();
-            mutationQueue.clear();
             storage.query(PendingMutation.PersistentRecord.class, Where.matchesAll(),
                 results -> {
+                    if (!results.hasNext()) {
```

Comment:
Unfortunate that we have to load all records simply to acquire a count. Can we add a `count` method to `LocalStorageAdapter`?

Reply:
Modified `load()` to only load one record and now updating the count of items as items are added or removed from the queue.